src/lib.rs

```diff
@@ -69,6 +100,36 @@ fn camera_zoom(
                 pos.translation = (mouse_world_pos
                     - mouse_normalized_screen_pos * proj_size * proj.scale)
                     .extend(pos.translation.z);
+
+                // Check whether the proposed camera movement would be within the provided boundaries, override it if needed
+                let scaling = Vec2::new(
+                    window.width() / (proj.right - proj.left),
+                    window.height() / (proj.top - proj.bottom),
+                ) * proj.scale;
+                if let Some(min_global_x) = cam.min_global_x {
+                    // Use the global pixel position provided to calculate the minimum camera x position allowable
+                    let min_cam_x = min_global_x + ((window.width() / 2.) * scaling.x);
+                    // Override the transform with the limit if needed. This prevents weird behavior with large mouse deltas
+                    pos.translation.x = pos.translation.x.max(min_cam_x);
+                }
+                if let Some(max_global_x) = cam.max_global_x {
+                    // Use the global pixel position provided to calculate the maximum camera x position allowable
+                    let max_cam_x = max_global_x - ((window.width() / 2.) * scaling.x);
+                    // Override the transform with the limit if needed. This prevents weird behavior with large mouse deltas
+                    pos.translation.x = pos.translation.x.min(max_cam_x);
+                }
+                if let Some(min_global_y) = cam.min_global_y {
+                    // Use the global pixel position provided to calculate the minimum camera y position allowable
+                    let min_cam_y = min_global_y + ((window.height() / 2.) * scaling.y);
+                    // Override the transform with the limit if needed. This prevents weird behavior with large mouse deltas
```

Comment:
This is to prevent us from zooming out over the border so it becomes visible, right? Could you maybe change the comment to explain that?